.github/workflows/deploy-demo-test.yml

```diff
@@ -0,0 +1,80 @@
+# Simple workflow for deploying static content to GitHub Pages
+name: Deploy static content to Pages
+
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["demo-test"]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python '3.9'
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'       
+
+      - name: Cache python packages for Linux
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ubuntu-latest-python-3.9            
+
+      - name: Install Kedro and other Python Dependencies
+        uses: "./.github/actions/install_kedro_and_python_dependencies"
+
+      - name: Setup Node.js and Install Dependencies
+        uses: "./.github/actions/install_node_dependencies"
+
+      - name: Build React application
+        run: |-
+          node -v
+          make build
+
+      - name: Install Kedro-Viz development package
+        run: |
+          pip install -e package
+        shell: bash
+
+      - name: Update demo-project build directory wit the latest changes
+        run: |
+          cd demo-project
+          if ! (kedro viz build |& tee /dev/stderr | grep -i -q "Success!"); then
+            exit 1
+          fi
+        shell: bash
+        
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v5
+      - name: Upload artifact
+        uses: actions/upload-pages-artifact@v3
+        with:
+          # Upload entire repository
+          path: 'demo-project/build'
+      - name: Deploy to GitHub Pages
+        id: deployment
```

Comment:
Is it possible to reuse https://github.com/kedro-org/publish-kedro-viz ?

Reply:
Yup, plan is to eventually move this to publish-kedro-viz. For now, I want to update the demo site without doing a full release, as there are some minor, non-user-facing features (all static content) I want to push. Before the release, I'll create another PR to ensure it creates build from publish-kedro-viz which is latest release version of kedro-viz instead of the latest development version.